content/blog/2022-12-18-please-test-black-23.1a1.md

```diff
@@ -0,0 +1,616 @@
+     """
+-    Remove punctuation and return cleaned string, in addition to its length in tokens."""
++    Remove punctuation and return cleaned string, in addition to its length in tokens.
++    """
+     pass
+```
+
+Although, Black won't move quotes of single line docstrings (as such with `example_2()`)
+since that would look ugly.
+
+This was [added to the preview style in 22.8.0][#3044] and
+[then tweaked in 23.1a1][#3430].
+
+### Better management of parentheses
+
+Here's a set of changes that together improve Black's handling of parentheses in various
+situations.
+
+<!-- 6, 7, 8, 11, 12 -->
+
+#### Return type annotations
+
+**Source:**
+
+```python3
+def foo() -> (int):
+    return 2
+
+
+def foo() -> intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasdskdsdsfdsafdsafsdfdasfffsfdsfdsafafhdskfhdsfjdslkfdlfsdkjhsdfjkdshfkljds:
+    return 2
+
+
+def frobnicate() -> ThisIsTrulyUnreasonablyExtremelyLongClassName | list[ThisIsTrulyUnreasonablyExtremelyLongClassName]:
+    pass
+```
+
+**Before:**
+
+```python3
+def foo() -> (int):
+    return 2
+
+
+def foo() -> intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasdskdsdsfdsafdsafsdfdasfffsfdsfdsafafhdskfhdsfjdslkfdlfsdkjhsdfjkdshfkljds:
+    return 2
+
+
+def frobnicate() -> ThisIsTrulyUnreasonablyExtremelyLongClassName | list[
+    ThisIsTrulyUnreasonablyExtremelyLongClassName
+]:
+    pass
+```
+
+**After** ([22.6.0+][#2990]):
+
+```python3
+def foo() -> int:
+    return 2
+
+
+def foo() -> (
+    intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasdskdsdsfdsafdsafsdfdasfffsfdsfdsafafhdskfhdsfjdslkfdlfsdkjhsdfjkdshfkljds
+):
+    return 2
+
+
+def frobnicate() -> (
+    ThisIsTrulyUnreasonablyExtremelyLongClassName
+    | list[ThisIsTrulyUnreasonablyExtremelyLongClassName]
+):
+    pass
+```
+
+#### Remove redundant parentheses around awaited objects
+
+**Source:**
+
+```python3
+async def main():
+    await (asyncio.sleep(1))
+    await (set_of_tasks | other_set)
+```
+
+**Before:** unchanged.
+
+**After** ([22.6.0+][#2991]):
+
+```python3
+async def main():
+    await asyncio.sleep(1)
+    await (set_of_tasks | other_set)
+```
+
+#### Remove redundant parentheses in with statements
+
+**Source:**
+
+```python3
+with (open("bla.txt")):
+    pass
+
+with (open("bla.txt")), (open("bla.txt")):
+    pass
+
+with (open("bla.txt") as f):
+    pass
+
+with (open("bla.txt")) as f:
+    pass
+```
+
+**Before:** unchanged.
+
+**After** ([22.6.0+][#2926]):
+
+```python3
+with open("bla.txt"):
+    pass
+
+with open("bla.txt"), open("bla.txt"):
+    pass
+
+with open("bla.txt") as f:
+    pass
+
+with open("bla.txt") as f:
+    pass
+```
+
+#### Remove redundant parentheses from except clauses
+
+**Source:**
+
+```python3
+try:
+    a.something
+except (AttributeError) as err:
+    raise err
+```
+
+**Before:** unchanged.
+
+**After** ([22.3.0+][#2939]):
+
+```python3
+try:
+    a.something
+except AttributeError as err:
+    raise err
+```
+
+#### Remove redundant (outermost) parentheses in for statements
```

Comment:
In my experience looking at a few code bases, this is the most impactful of the parens management changes. Might be worth calling out?

Reply:
Good idea. I also highlighted "remove blank lines after code block open" since that seems like a high-impact one too.